go/adbc/driver/snowflake/connection.go

```diff
@@ -557,13 +587,204 @@ func (c *cnxn) populateMetadata(ctx context.Context, depth adbc.ObjectDepth, cat
+			if err != nil {
+				return nil, errToAdbcErr(adbc.StatusIO, err)
+			}
+		}
+
+		if _, exists := availableConstraintTypes[internal.ForeignKey]; exists {
+			foreignKeyConstraintsData, err = c.getForeignKeyConstraints(ctx, availableFullyQualifiedConstraints)
+			if err != nil {
+				return nil, errToAdbcErr(adbc.StatusIO, err)
+			}
+		}
+	}
+
+	tableConstraintsData := append(append(uniqueConstraintsData, primaryKeyConstraintsData...), foreignKeyConstraintsData...)
+
+	sort.Slice(tableConstraintsData, func(i, j int) bool {
+		if tableConstraintsData[i].constraintName == tableConstraintsData[j].constraintName {
+			return tableConstraintsData[i].keySequence < tableConstraintsData[j].keySequence
+		}
+		return tableConstraintsData[i].constraintName == tableConstraintsData[j].constraintName
+	})
+
+	return tableConstraintsData, nil
+}
+
+func (c *cnxn) getUniqueConstraints(ctx context.Context, fullyQualifiedConstraints map[string]bool) ([]TableConstraint, error) {
+	uniqueConstraintsData := make([]TableConstraint, 0)
+
+	rows, err := c.sqldb.QueryContext(ctx, prepareUniqueConstraintSQL(), nil)
+	if err != nil {
+		return nil, errToAdbcErr(adbc.StatusIO, err)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var uniqueConstraint TableConstraint
+		if err := rows.Scan(&uniqueConstraint.skipCreatedOn, &uniqueConstraint.dbName, &uniqueConstraint.schema, &uniqueConstraint.tblName, &uniqueConstraint.colName, &uniqueConstraint.keySequence, &uniqueConstraint.constraintName, &uniqueConstraint.skipRely, &uniqueConstraint.skipComment); err != nil {
+			return nil, errToAdbcErr(adbc.StatusInvalidData, err)
+		}
+
+		currentConstraintQualifiedName := getFullyQualifiedConstraintName(uniqueConstraint.dbName, uniqueConstraint.schema, uniqueConstraint.tblName, uniqueConstraint.constraintName)
+
+		// skip constraint if it doesn't exist in fullyQualifiedConstraints
+		if _, exists := fullyQualifiedConstraints[currentConstraintQualifiedName]; exists {
+			uniqueConstraint.constraintType = internal.Unique
+			uniqueConstraintsData = append(uniqueConstraintsData, uniqueConstraint)
+		}
+
+	}
+	return uniqueConstraintsData, nil
+}
+
+func (c *cnxn) getPrimaryKeyConstraints(ctx context.Context, fullyQualifiedConstraints map[string]bool) ([]TableConstraint, error) {
+	primaryKeyConstraintsData := make([]TableConstraint, 0)
+
+	rows, err := c.sqldb.QueryContext(ctx, preparePrimaryKeyConstraintSQL(), nil)
+	if err != nil {
+		return nil, errToAdbcErr(adbc.StatusIO, err)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var primaryKeyConstraint TableConstraint
+		if err := rows.Scan(&primaryKeyConstraint.skipCreatedOn, &primaryKeyConstraint.dbName, &primaryKeyConstraint.schema, &primaryKeyConstraint.tblName, &primaryKeyConstraint.colName, &primaryKeyConstraint.keySequence, &primaryKeyConstraint.constraintName, &primaryKeyConstraint.skipRely, &primaryKeyConstraint.skipComment); err != nil {
```

Comment:
split the line to make this easier to read

Reply:
split into multiple lines